src/components/common/UserProfileEdit/UserMainSettings.tsx

```diff
@@ -113,57 +86,35 @@ const UserMainSettings = ({
   };
 
   const handleSubmit = (updatedProfile: FormValues) => {
-    const valuesToSubmit = {
-      email: updatedProfile.email === '' ? null : updatedProfile.email,
-    };
-
     editUser({
       variables: {
         input: {
           id: walletAddress,
-          ...userProfile,
           ...updatedProfile,
-          ...valuesToSubmit,
+          email: updatedProfile.email || null,
         },
       },
     });
   };
 
-  useEffect(() => {
-    if (called && !loading && updateUser) {
-      updateUser(walletAddress);
-      setShowSnackbar(true);
-    }
-  }, [called, loading, walletAddress, updateUser]);
-
+  useEffect(
+    // update user on dismount
+    () => () => {
+      updateUser?.(walletAddress);
+    },
+    [walletAddress],
+  );
   return (
     <>
-      <Heading
-        appearance={{ theme: 'dark', size: 'medium' }}
-        text={MSG.heading}
-      />
+      <UserInfo user={user} />
       <Form<FormValues>
         defaultValues={defaultValues}
         onSubmit={handleSubmit}
         validationSchema={validationSchema}
+        resetOnSubmit
       >
-        {({ formState: { isSubmitting, isDirty, isValid } }) => (
+        {({ formState: { isDirty, isValid } }) => (
           <div className={styles.main}>
-            <FieldSet>
-              <InputLabel label={MSG.labelWallet} />
```

Comment:
"UserInfo" is more descriptive in my opinion as to what this portion of the code is doing.